Automattic/wp-calypso
# Pattern Library: Fix scroll bug when loading single-pattern permalink

Fixes https://github.com/Automattic/dotcom-forge/issues/6251

## Proposed Changes

This PR addresses the following scroll-related issues:
- Loading single-pattern permalink yields a janky scroll experience (the page jumps up and down during loading).
- The window scroll doesn't reset when navigating between categories.

One issue with the previous logic was the use of `Element.scrollIntoView` in the category pill navigation. `scrollIntoView` doesn't just scroll the parent on the X axis but also the entire viewport on the Y axis, which is bad when loading single pattern permalinks.

I've also removed the `id` attributes from the `.pattern-preview` elements because the native browser experience of scrolling to an element based on the URL hash greatly contributed to the janky loading experience. Instead, we now wait for the patterns to load, plus a 1-second timeout, and then call `scrollIntoView` on the relevant `.pattern-preview`.

Lastly, I've made it so that when users navigate between categories, we scroll back up to the "anchoring position" of the sticky category pill navigation. The scroll doesn't change when navigating between categories if the user hasn't scrolled below that point.

This screencast shows all these behaviors in action:


https://github.com/Automattic/wp-calypso/assets/1101677/cb05e9d3-a992-40f6-a891-3ad830edde12



## Testing Instructions



1. Navigate to `/patterns/about`
2. Scroll down the page for a bit
3. Copy the permalink to a pattern
4. Open a new tab, paste and go to the link
5. Ensure that the page scrolls down to the relevant pattern and doesn't scroll back up
6. Click on a category in the category pill navigation
7. Ensure that the page scrolls back up to the "anchoring position" of the sticky category pill navigation
